lib/runtime-core/src/backend.rs

```diff
@@ -200,6 +200,15 @@ pub struct Features {
     pub threads: bool,
 }
 
+#[derive(Debug)]
+pub struct BackendCompilerConfig(pub Box<dyn Any + 'static>);
```

Comment:
Some doc comments and maybe some implementation-specific comments would be good to have here -- this code is a bit abstract and can be scary

Reply:
I added some, let me know if you think I got it or if there's more to add / something I forgot.